metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -687,6 +687,9 @@ class TableauSourceReport(
     num_hidden_assets_skipped: int = 0
     logged_in_user: List[UserInfo] = dataclass_field(default_factory=list)
 
+    def compute_stats(self) -> None:
+        self.close_stage()
```

Comment:
this doesn't seem right?

`compute_stats()` is called whenever we print out stats, which could happen in the middle of a stage

Reply:
That's what I meant in PR description by "abusing" `compute_stats`. While it worked, or at least it worked for my test with Tableau, I agree you.

In commit 2fe9a1c I have reverted updates on the `IngestionStageReport` implemented a new `IngestionStageContextReport` based on `ContextManager`. So far, only used by Tableau ingestor. If we agree on the approach here, in follow up PRS I can update other connectors to use the new implementation and remove the old one